packages/edit-site/src/components/block-editor/use-navigate-to-entity-record.js

```diff
@@ -2,26 +2,76 @@
  * WordPress dependencies
  */
 import { privateApis as routerPrivateApis } from '@wordpress/router';
-import { useCallback } from '@wordpress/element';
+import { useCallback, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { unlock } from '../../lock-unlock';
 
-const { useHistory } = unlock( routerPrivateApis );
+const { useHistory, useLocation } = unlock( routerPrivateApis );
 
 export default function useNavigateToEntityRecord() {
 	const history = useHistory();
+	const { getSelectedBlockClientId } = useSelect( ( select ) => {
+		const { getSelectedBlockClientId: getSelected } =
+			select( blockEditorStore );
+		return {
+			getSelectedBlockClientId: getSelected,
+		};
+	}, [] );
```

Comment:
Any reason for not using a static getter here as well?

```
const { getSelectedBlockClientId } = useSelect( blockEditorStore );
```

Reply:
Oh, I could have sworn I'd updated that 😅 lemme fix it now